cpp/src/bitmask/null_mask.cu

```diff
@@ -168,13 +169,14 @@ __global__ void count_set_bits_kernel(bitmask_type const *bitmask,
  * @param source The mask to copy from
  * @param source_begin_bit The offset into `source` from which to begin the copy
  * @param source_end_bit   The offset into `source` till which copying is done
- * @param number_of_mask_words The number of words of type bitmask_type to copy
  *---------------------------------------------------------------------------**/
 __global__ void copy_offset_bitmask(bitmask_type *__restrict__ destination,
                                     bitmask_type const *__restrict__ source,
                                     size_type source_begin_bit,
-                                    size_type source_end_bit,
-                                    size_type number_of_mask_words) {
+                                    size_type source_end_bit) {
+  auto number_of_mask_words = cudf::util::div_rounding_up_safe(
```

Comment:
I didn't realize this is what it would take to remove the `number_of_mask_words` argument from the kernel. I'd honestly prefer just reverting that change.

Besides, doesn't this function throw an exception? How can it be called from device code?

Reply:
Reverted.